pipenv/utils.py

```diff
@@ -572,6 +571,10 @@ def convert_deps_from_pip(dep):
             dependency[req.name] = extras
             if specs:
                 dependency[req.name].update({'version': specs})
+        if hasattr(req, 'markers'):
+            if isinstance(dependency[req.name], six.string_types):
```

Comment:
just for my knowledge, how do you hit this edge?

Reply:
Lol. So above this we basically allow a pipfile entry to wind up in the state of `name = “==x.y”` if all we have is a version. This piece just dict-ifies that case if we have markers to add.

Markers is not always an attribute on a requirement however. I tried a few other approaches without success